test/w3c/w3ctest.js

```diff
@@ -50,40 +52,21 @@ function createJsdom(source, url, t) {
       };
     },
 
-    loaded: function (err) {
-      t.ifError(err && err[0].data.error);
+    loaded(err) {
+      t.ifError(err);
     }
   });
 }
 
-function testUrl(url, t) {
-  fs.readFile(path.resolve(__dirname, "tests", url), "utf8", function (err, file) {
-    if (err) {
-      request.get("http://w3c-test.org/" + url, function (err, resp, respBody) {
-        if (err) {
-          t.ifError(err, "request should go through without error");
-          t.done();
-          return;
-        }
-        if (resp.statusCode < 200 || resp.statusCode >= 300) {
-          t.ok(false, "request should return OK status code");
-          t.done();
-          return;
-        }
-
-        createJsdom(respBody, "http://w3c-test.org/" + url, t);
-      });
-    } else {
-      file = file.replace(/\/resources\//gi, toFileUrl(__dirname + "/tests/resources") + "/");
-      createJsdom(file, toFileUrl(path.resolve(__dirname, "tests", url)), t);
-    }
-  });
-}
+const server = http.createServer(ecstatic({ root: path.resolve(__dirname, "tests") })).listen(port, hostname);
```

Comment:
I'm not terribly comfortable with this, but couldn't think of anything that was much better. Suggestions welcome :-/.